src/runtime/auto-prompt-manager.js

```diff
@@ -436,6 +457,22 @@ export class AutoPromptManager {
       return false;
     }
 
+    // If there is an audience action with explicit higher priority than the contribution
+    // or subscription action, do not show the autoprompt.
+    const autopromptActionType = this.isSubscription_({autoPromptType})
+      ? TYPE_SUBSCRIPTION
+      : this.isContribution_({autoPromptType})
+      ? TYPE_CONTRIBUTION
+      : null;
+    if (autopromptActionType) {
+      const autopromptActionIndex = actions.findIndex(
+        (action) => action.type === autopromptActionType
+      );
+      if (autopromptActionIndex > 0) {
+        return false;
```

Comment:
What if the non-autoprompt action ended up suppressed in `getAudienceActionPromptType_`? I think it can result in `shouldShowAutoPrompt = false` even when contribution should be shown. I have a concern that it could be misleading for future reference, so would rather prefer not covering this case as there's no defined behavior for when an audience action can have explicit higher priority.
I feel like respecting priority (for autoprompt type) from audience engine should be handled within `getAudienceActionPromptType_` once we can remove the customized logic (e.g., show contribution first whenever it's availalbe) and delegate triggering decision to the BE.

Reply:
Good point, though we currently do cover the case where an audience action has an explicit higher priority than contribution: https://github.com/subscriptions-project/swg-js/blob/main/src/runtime/auto-prompt-manager-test.js#L1916

autoprompt suppression logic:
https://github.com/subscriptions-project/swg-js/pull/2934/files/9d4e5bd1be0b9c64169d45bec880407f5601f00f#diff-c74361e6f3e0a3fc5042519e0ddf35b67dc861e52b588913d6ebcf696ff17f12L559-L581

This was added as a part of b/270965538, https://github.com/subscriptions-project/swg-js/pull/2820/files

If we want to maintain this, then `getAudienceActionType_` should not filter out autoprompttypes from potential actions.

Reply:
Spoke offline, `getAudienceActionType_` will revert to check for contribution action.